File: 2ndSemester/kursprogramowania/lista3/cpp/src/hexagon.cpp
#include <hexagon.hpp>

#include <cmath>
#include <string>

Hexagon::Hexagon(double side) noexcept(false)
{
    if (side <= 0)
        throw std::invalid_argument(std::string(__FILE__) + ":" + std::to_string(__LINE__) + " [" + __func__ + "] " + "side should be positive, got " + std::to_string(side));

    this->side = side;
}

double Hexagon::area() const noexcept(true) 
{
    return 2.5980762113 * this->side * this->side;
}

double Hexagon::perimeter() const noexcept(true)
{
    return 6 * this->side;
}

const std::string Hexagon::name() const noexcept(true) 
{
    return "Hexagon";
}

File: 2ndSemester/kursprogramowania/lista3/java/Hexagon.java
/**
 * Hexagon
 */
public final class Hexagon extends Figure 
{
    private final double side;
    
    public Hexagon(final double side)
    {
        if (side <= 0) 
        {
            throw new IllegalArgumentException("Side length should be positive, got " + side);    
        }
        this.side = side;
    }

    @Override
    public double area() 
    {
        return 2.5980762113 * side * side;
    } 

    @Override
    public double perimeter()
    {
        return 6 * side;
    }

    @Override
    public String name()
    {
        return "Hexagon";
    }
}
